File: app/src/main/java/com/raredev/vcspace/ui/widget/NavigationSpace.java
package com.raredev.vcspace.ui.widget;

import android.app.Activity;
import android.content.Context;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.*;

import com.google.android.material.textview.MaterialTextView;
import com.raredev.vcspace.R;

public class NavigationSpace extends LinearLayout {
    
    public NavigationSpace(Activity context) {
        super(context);
        init();
    }
    
    public NavigationSpace(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }
    
    public void init() {
        setOrientation(HORIZONTAL);
        setGravity(Gravity.CENTER);
    }
    
    public void addItem(Activity activity, CharSequence title, int icon, View.OnClickListener listener) {
        LayoutInflater inflater = activity.getLayoutInflater();
        View view = inflater.inflate(R.layout.navigation_item, this, false);
        view.setOnClickListener(listener);
        view.setTooltipText(title);
        
        ImageView imgIcon = view.findViewById(R.id.icon);
        imgIcon.setImageResource(icon);
        
        MaterialTextView tvTitle = view.findViewById(R.id.title);
        tvTitle.setText(title);
        
        addView(view);
    }
    
    public void clear() {
        removeAllViews();
    }
}

File: app/src/main/java/com/teixeira/vcspace/ui/NavigationSpace.kt
package com.teixeira.vcspace.ui

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.LinearLayout
import com.teixeira.vcspace.databinding.LayoutNavigationItemBinding

class NavigationSpace : LinearLayout {

  constructor(context: Context) : super(context)

  constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

  constructor(
    context: Context,
    attrs: AttributeSet?,
    defStyleAttr: Int
  ) : super(context, attrs, defStyleAttr)

  fun addItem(title: Int, icon: Int, listener: OnClickListener) {
    addItem(context.getString(title), icon, listener)
  }

  private fun addItem(title: CharSequence, icon: Int, listener: OnClickListener) {
    val binding = LayoutNavigationItemBinding.inflate(LayoutInflater.from(context))
    binding.root.setOnClickListener(listener)
    binding.root.tooltipText = title
    binding.icon.setImageResource(icon)
    binding.title.text = title
    addView(binding.root, LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT, 1f))
  }
}
